error[E0603]: trait `StrExt1` is private
  --> tests/ui/visibility.rs:37:14
   |
37 |     use foo::StrExt1; //~ ERROR trait `StrExt1` is private [E0603]
   |              ^^^^^^^ private trait
   |
note: the trait `StrExt1` is defined here
  --> tests/ui/visibility.rs:7:5
   |
 7 |     impl str {
   |     ^^^^^^^^

error[E0603]: trait `StrExt2` is private
  --> tests/ui/visibility.rs:39:14
   |
39 |     use foo::StrExt2; //~ ERROR trait `StrExt2` is private [E0603]
   |              ^^^^^^^ private trait
   |
note: the trait `StrExt2` is defined here
  --> tests/ui/visibility.rs:14:5
   |
14 |     pub(self) impl str {
   |     ^^^^^^^^^^^^^^^^^^

error[E0603]: trait `StrExt3` is private
  --> tests/ui/visibility.rs:41:19
   |
41 |     use foo::bar::StrExt3; //~ ERROR trait `StrExt2` is private [E0603]
   |                   ^^^^^^^ private trait
   |
note: the trait `StrExt3` is defined here
  --> tests/ui/visibility.rs:24:9
   |
24 |         pub(super) impl str {
   |         ^^^^^^^^^^^^^^^^^^^
